pkg/ccl/backupccl/restore_planning.go

```diff
@@ -1720,6 +1902,10 @@ func doRestorePlan(
 		if err := dropDefaultUserDBs(ctx, p.ExecCfg()); err != nil {
 			return err
 		}
+
+		if err := moveConflictingSystemTables(ctx, p, databasesByID, schemasByID, filteredTablesByID, typesByID); err != nil {
```

Comment:
This is a pretty heavy operation to have in planning, but for now I wanted to make sure this work is completely done before we start thinking about rewrites.